### ISSUE TITLE
Events not behaving as expected

### ISSUE THREAD

**Issue by Sheepyhead** [PradeepKumarRajamanickam/kurinji#36]
I'm using bevy 0.3.0 and kurinji 1.0.1, and I was attempting to implement a proof of concept using events to register the following flow:
- When the user clicks the left mouse button: A single message is printed indicating that the mouse button has just been pressed down
- When the user holds the left mouse button: After the prior message, a different message is repeated for as long as the button is held
- When the user lets go of the left mouse button: A single message says that the mouse button has been let go and no further messages appear until the button is clicked again

I did this by looking at the event example from the repo. I implemented a struct with three eventreaders, for OnActionBegin, OnActionActive and OnActionEnd. Then I added three systems, one for each reader, that take the entire struct in and attempts to read the latest action to see if it is what is needed. Unfortunately instead of the expected behavior;
- mouse button clicked
- mouse button held
- mouse button held
...
- mouse button released

What I see is the following:
- mouse button clicked
- mouse button held
- mouse button clicked
- mouse button held

Repeated until I release the mouse button with no trace of my OnActionEnd message. The following is my code for the event reader struct and the systems

```rust
#[derive(Default)]
pub struct ActionState {
    begin_reader: EventReader<OnActionBegin>,
    active_reader: EventReader<OnActionActive>,
    end_reader: EventReader<OnActionEnd>,
}

fn player_action_begin(
    mut state: Local<ActionState>,
    action_begin_event: Res<Events<OnActionBegin>>,
    mut query: Query<&mut Player>,
) {
    let action = match state.begin_reader.latest(&action_begin_event) {
        Some(value) => value,
        None => return,
    };
    let player = match query.iter_mut().next() {
        Some(value) => value,
        None => return,
    };

    if action.action == "MOVE_TO_MOUSE" {
        println!("{:?} MOVE TO MOUSE", SystemTime::now());
    }
}

fn player_action_active(
    mut state: Local<ActionState>,
    action_active_event: Res<Events<OnActionActive>>,
    mut query: Query<&mut Player>,
) {
    let action = match state.active_reader.latest(&action_active_event) {
        Some(value) => value,
        None => return,
    };
    let player = match query.iter_mut().next() {
        Some(value) => value,
        None => return,
    };

    if action.action == "MOVE_TO_MOUSE" {
        println!("{:?} keep moving to mouse", SystemTime::now());
    }
}

fn player_action_end(
    mut state: Local<ActionState>,
    action_end_event: Res<Events<OnActionEnd>>,
    mut query: Query<&mut Player>,
) {
    let action = match state.end_reader.latest(&action_end_event) {
        Some(value) => value,
        None => return,
    };

    let player = match query.iter_mut().next() {
        Some(value) => value,
        None => return,
    };

    if action.action == "MOVE_TO_MOUSE" {
        println!("{:?} stop moving TO MOUSE", SystemTime::now());
    }
}
```

Big fan of this plugin, and I hope I can get it to work the way I wanted!

**Comment by PradeepKumarRajamanickam** [comment]
Hey @Sheepyhead, Glad you like the plugin. Yes there is something definitely wrong with the events you are receiving. I will check it out.

**Comment by PradeepKumarRajamanickam** [comment]
I have published a patch release on crate.io. @Sheepyhead Can you try it and let me know if it solved your issue.

**Kurinji v1.0.3**
Changelog
- fixed* issue Events not behaving as expected #36
- fixed* not all action's prev str are cached 
- fixed* on action end not fired

**Comment by Sheepyhead** [comment]
> 
> 
> I have published a patch release on crate.io. @Sheepyhead Can you try it and let me know if it solved your issue.
> 
> **Kurinji v1.0.3**
> Changelog
> 
>     * fixed* issue Events not behaving as expected #36

I can confirm that 1.0.3 now behaves exactly as expected! Thanks so much for the quic response

**Comment by PradeepKumarRajamanickam** [comment]
closing ticket
